store/models.py

```diff
@@ -567,6 +567,13 @@ class OptionProduct(BaseProduct):
         default=False,
     )
 
+    available_on_retreat_types = models.ManyToManyField(
+        "retirement.RetreatType",
+        verbose_name=_("Applicable retreat types"),
+        related_name='option_products',
+        blank=True,
+    )
+
```

Comment:
Maybe it would have been better to add it the same way as the already existing ?: https://github.com/FJNR-inc/Blitz-API/blob/2547f4615a3742a06eefdec44e1f24a652026547/store/models.py#L377-L390

Reply:
I thought too, but in your case we are doing a generic FK because we have several type of product. In our case we want a field dedicated to RetreatType so that's why I chose a M2M linked to RetreatType